### ISSUE TITLE
CRM / Returns

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1037]
_Imported from trac ticket 1037.
Created by **rneville**
Opened in trac: Aug 30, 2011, 11:06:09 AM
Last modified in trac: Jun 4, 2013, 12:47:34 PM_
***
_by rneville:_
See design [here](comment:27)
----
## Not Implemented with this (phase 1) ticket
 * Financing implications with returns ([#2224](2224))
 * Writing returns against open order with RF enabled ([#2295](2295))
 * Writing returns for more than was originally sold with RF enabled ([#2295](2295))
 * Returns with an Outlet status different from original line ([#2327](2327))
 * Closing return lines to new NAS or outlet status ([#2327](2327), [#2328](2328))
 * Closing returns for more than was originally sold (figuring cost issue) ([#2329](2329))
 * Closing returns with V12 ([#2317](2317))
 * Closing returns with RF ([#2295](2295))
 * Linking to installment contracts, ([#2281](2281))
 * Writing returns with no adjusted order (from scratch), ([#2291](2291))

***
* kbower attached [CMF.lst](https://trac.retailarchitects.com/trac/attachment/ticket/1037/CMF.lst) on May 9, 2013, 1:48:52 PM
* kbower attached [crmoutlet.lst](https://trac.retailarchitects.com/trac/attachment/ticket/1037/crmoutlet.lst) on May 9, 2013, 1:49:26 PM
* kbower attached [crmtoomany.lst](https://trac.retailarchitects.com/trac/attachment/ticket/1037/crmtoomany.lst) on May 9, 2013, 1:49:44 PM
* kbower attached [inv_xref_arc0729230GHLD.lst](https://trac.retailarchitects.com/trac/attachment/ticket/1037/inv_xref_arc0729230GHLD.lst) on May 9, 2013, 1:50:16 PM
* kbower attached [spiff_for_orphan_return.txt](https://trac.retailarchitects.com/trac/attachment/ticket/1037/spiff_for_orphan_return.txt) on May 10, 2013, 1:55:48 PM

**Comment by kbower** [comment]
_**rneville** trac comment on Aug 30, 2011, 11:13:04 AM:_

 * creating a CRM w/o an invoice is permission controlled

**Comment by kbower** [comment]
_**rneville** trac comment on Aug 30, 2011, 11:26:37 AM:_

 * delivery date uses DEL table like normal SAL does.

**Comment by kbower** [comment]
_**rneville** trac comment on Aug 30, 2011, 11:42:53 AM:_

 * CRM gets a whole new document number. ~~Artrans write adj. invoice as original ticket number~~

_Edit: stricken out sentence is exactly wrong._

**Comment by kbower** [comment]
_**rneville** trac comment on Oct 14, 2011, 11:08:43 AM:_

Hi Guys,
 
One thing that came up today at Steinhafels is credit memos.  In GERS today credit memos only have one cause code for the entire order.  We would like to be able to enter a cause code for each line on the credit memo. This would be like a service order today. Where there is a cause code for each line on a service order.  Just something for you to think about when designing your new system and credit memos.

-- 
Thanks,

Doug Warner
----
moved to new ticket: [#2278](2278)

**Comment by kbower** [comment]
_**kbower** trac comment on Jun 18, 2012, 8:21:55 AM:_

see #718

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 26, 2012, 5:30:36 PM:_

# Spiffs

_Read from bottom up_
----
I would agree with the following statement: “I suspect a better solution is based on the credit rendered to the customer for the return... if the customer is returned more than the minimum then the full SPIFF is applied, but if the customer is returned less than the minimum then $0 SPIFF on return”

 

As for the question on whether the MCR/MDB is attached to a SAL or not, here are my results: 

 

If the ORIGINAL order is attached to the MCR/MDB, then the entire SPIFF that appears in the Spiff Calendar will be reflected on each SO_LN (no matter how much SPIFF was actually in the original SO).  Even if you change the prices on the SKU, the SPIFF will NOT change.

 

BUT, if there is NO original order entered (ie, MCR/MDB is not attached to anything – which requires security to be able to create), NO SPIFF will be attached to the lines.  You will just be credited or debited for the retail amounts, without having any SPIFF on the SO_LN of the MCR/MDB.

 

BTW:  Thanks for the attention.

 

Jesús R. Capó, VP, CIO

----
From: Kent Bower [mailto:kb@retailarchitects.com]
Sent: Wednesday, July 25, 2012 5:57 PM
To: Jesus Capo
Cc: Mariam Farach; Sal Genovese; WayneLiburd; Michelle Chadwick; dev
Subject: Re: spiffs for MCR, MDB, CRM, EEX

 

Once again, thank you JR! (I'll be sure to give you more attention than Kane's... just kidding! Wayne is extraordinarily helpful with my questions as well)

Does it matter whether the MCR/MDB is attached to a SAL or not?

Regarding credit memos, I'm not surprised that GERS gets out of sync from the SAL line.  Is that the goal, to keep these in sync?  If so, I don't understand why an MCR should be different from a CRM???  Unfortunately, we might too, because, as discussed at the last sprint, there is no true value in linking the SAL line to the CRM line.... you can't tell which one the customer returned anyway in the case of qty > 1.  But without such a link, there is no way to keep a SPIFF in sync.

I also wonder if we need to keep it in sync... can we instead look at the original SAL when deducting SPIFFs?  Alternatively to keeping them in sync we could average them, which will 99% of the time work out to exactly what you wanted.

As an example:
Customer bought 4 chairs, one was discounted for some reason (coupon or as is), so there was a $10 spiff on 3 of them but $0 on the 4th.  The customer returns one chair (we really have no real way to know which one if it was already in the home for a little while), so what we could do is average it and say, well, they returned 1 of 4 chairs, so the SPIFF on the CRM is 1/4th of the total $30, or $7.50.

I suspect a better solution is based on the credit rendered to the customer for the return... if the customer is returned more than the minimum then the full SPIFF is applied, but if the customer is returned less than the minimum then $0 SPIFF on return... I think that is likely what GERS was trying to accomplish, right?


----


On 7/25/2012 5:42 PM, Jesus Capo wrote:

Here are my results (using GERS, and not LOFT):

 

Setting #33 to Y works.  The SPIFF is removed if the retail falls below the minimum.

 

MCR or MDB entered has ZERO logic.  The SPIFF will always appear no matter what retail amount is entered.

 

CRM entered carries over the original SPIFF amounts that appear on the original referencing Sales Order  BUT  If the CRM retail is adjusted to either BELOW or ABOVE the minimum retail previously setup in the calendar, then the CRM uses the Sales Order minimum retail logic and adjusts the SPIFF accordingly.  This really doesn’t make any sense because if the original order had ZERO Spiff paid because it was BELOW the minimum retail, and for some reason I enter a Credit Memo that has a Retail HIGHER than the minimum, then the SPIFF will appear on the Credit Memo, thus Crediting a SPIFF that was never Debited in the first place.

 

Is this understandable?

 

Jesús R. Capó, VP, CIO

----


From: Kent Bower [mailto:kb@retailarchitects.com]
Sent: Wednesday, July 25, 2012 4:55 PM
To: Jesus Capo
Cc: Mariam Farach; Sal Genovese; WayneLiburd; Michelle Chadwick; dev
Subject: Re: spiffs for MCR, MDB, CRM, EEX

 

Again, thank you.  The case I'm wondering is if your parameter 33. were set to "Y" and you added a minimum retail for SPIFF for a SKU and then added that SKU to an MCR or MDB or CRM at a price below that minimum retail, does the SPIFF get recorded on the MCR line? 

Any chance you would test this for me (I'd rather have it correct from the start for MCR/MDB/CRM than need to fix the bug later):

1. Set parameter 33 to Y
2. Pick a SKU and set a minimum retail for the SPIFF.
3. Pick an amount below that minimum retail, write a SAL line for that SKU at that below retail amount.
4. Verify that the SPIFF is not recorded on the SAL SO_LN.
5. Pick an amount above minimum and verify the SPIFF is recorded for a line of that price.
6. Write an MCR line for that same SKU at the same below minimum retail amount from step 3 above.  Is the SPIFF recorded on the MCR SO_LN?
7. Write an MCR line for the same SKU at the above minimum price from step 5 and verify SPIFF is recorded on MCR SO_LN.
8. Repeat step 6 & 7 for MDB & CRM
9. Does the answer to 6 change if the MCR is adjusting a SAL vs. an ad hoc MCR?

P.S. We didn't know there was such a parameter 33 and Loft is coded as if the answer is 'Y', but as long as you add no minimum retail to your spiff setup, it works out the same.  So don't do that, ok?  (We can inspect the parameter value if needed, have you had issues with this in Loft?)

----

On 7/25/2012 4:07 PM, Jesus Capo wrote:

I am not sure I understand the question, but here is what I've found:

 

The following  system parameter is in GERS:

 

33. Require minimum retail to pay SPIFF in RES, SOE, CCRC,PSOE, and SOM?                                              : N

34. If SKU is not on original sale or original sale is not on file, attach SPIFF amount to CRM line items in SOE & SOM?   Y

 

I've tested and no matter what amount is entered in the MCR, the SPIFF goes attached to the SKU on the MCR (no matter if it brings the minimum retail to below the established minimum in the SPIFF calendar).

 

I think there is no logic beyond the initial SPIFF calendar and minimum Retail at the time the Sales Order is entered.  After that there seems to be none.

 

 

Jesús R. Capó, VP, CIO

----
From: Kent Bower [mailto:kb@retailarchitects.com]

Sent: Tuesday, July 24, 2012 7:52 PM

To: Jesus Capo

Cc: Mariam Farach; Sal Genovese; WayneLiburd; Michelle Chadwick; dev

Subject: Re: spiffs for MCR, MDB, CRM, EEX

 

First, thank you JR.

 

Question: does it follow the exact same logic as sales lines, meaning if the amount of the MCR/MDB/CRM, etc line doesn't meet the spiff minimum then there is no spiff? (This doesn't seem logical, but my guess is GERS might just be using the exact same logic?)

 

Kent

 

----

On Jul 24, 2012, at 4:19 PM, Jesus Capo <Jesus@ElDoradoFurniture.com> wrote:

 

    I have a late response to your question.

     

    How MCR, MDB, CRM, etc. affects commissions depends on how the user company pays their commissions.  When the MCR/MDB is created, the SO_LN will carry over  the ENTIRE Spiff amount for that SKU, no matter what the MCR amount is for.    How that SPIFF is used depends on each company.

     

    At El Dorado, we pay commissions based on our own particular SQL formulas and usually do not take the MCR/MDB spiff into consideration, unless the ENTIRE retail amount is being removed.

     

    Does this help?

     

    Jesús R. Capó, VP, CIO


----

    From: Kent Bower [mailto:kb@retailarchitects.com]

    Sent: Tuesday, July 17, 2012 5:32 PM

    To: Jesus Capo; Mariam Farach; Sal Genovese; Wayne Liburd; Michelle Chadwick

    Cc: dev

    Subject: spiffs for MCR, MDB, CRM, EEX

     

    Will someone please take a few minutes to explain what meaning a spiff has for MCR, MDB, CRM and EEX and also how does gers populate them?

    Does GERS copy this from the line being adjusted?  Is the value prorated?  Is MDB different?  Is the spiff just looked up from the spiff calendar/ITM table?

     

    Thanks very much in advance,

    Kent

     

    P.S. Don't be shy.

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 18, 2012, 8:24:52 AM:_

 When a return is finalized it should show in the document what site and
location the item is returned to, but currently the view in LOFT does not
show that. That needs to be added to the LOFT screen


Thanks

Sal

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 18, 2012, 8:25:21 AM:_

Ticket [#1984](1984) has been marked as a duplicate of this ticket.

**Comment by kbower** [comment]
_**rneville** trac comment on Jan 11, 2013, 10:07:15 AM:_

pulling forward

**Comment by kbower** [comment]
_**sjohnson** trac comment on Jan 22, 2013, 11:24:58 AM:_

Scott,
We would only want a credit to be keyed for the original quantity purchased. As well, would this functionality allow returns from multiple split orders to one return tab? It would be great to see that a credit memo could only be keyed for merchandise on one split transportation at a time. This will clean up the back end with a credit memo amount not exceeding the amount of that specific delivery document number in CINQ.
Thanks,
Melissa

----

From: Scott Johnson [mailto:sj@retailarchitects.com]
Sent: Tuesday, January 22, 2013 11:13 AM
To: Melissa Coral
Cc: GOLIVE HELP; Brenda E. Gelbert; Johnathan Phelps
Subject: Re: Credit Memos in Loft

Melissa,
You could get a pretty accurate sense of how it would work by dragging a line to the Add Credit (MCR) tab.  Our intend would be to duplicate this functionality.  I think that makes sense in the case of CRM to only allow up to the quantity sold to be credited.
Scott

**Comment by kbower** [comment]
_**sjohnson** trac comment on Jan 29, 2013, 1:43:32 PM:_

Read from top down:
Dear Retailers,
Soon, we'll begin working on Credit Memos in Loft.  Here is just a quick question to get us started.  We know that Credit Memos are usually created against an existing order.  Is that 100% the case or do we need to add permission based ability to create a Credit Memo from scratch?
Thanks,
Scott

----
You may have to create a credit memo from scratch BUT it should be  permission based.

I remember having used this functionality a couple of times when I worked in customer service. Most of the time it was related to warranties on orders over 10 years old (mattresses)

This is the way GERS has it, permission based.
Hope it helps,
Mariam

----
From Andrea at Miskelly
From scratch...we do this on sales that no longer exist in our system due to archiving or other reason. It is definitely the exception, but we do need functionality.

----
From Sal
I can’t think of a scenario where you would return an item without an original sales order already existing, unless the original is archived. It would be rare but I think a security to do it would be needed at some point.

From Sharon Ford at Ivan
Is there a way to force the system to look at the delivery fee on the original sale and use that amount on the CRM?  Store 49 sale 0123349YU53 had a delivery fee of $49.99, but CRM used $79.99 – This has always been an issue in GERS, but thought maybe LOFT would be different              

----
From Doug Warner
We need the ability to create a Credit Memo from scratch.  If the sale is archived you can't link it to the original sale.

----
Darlene at Morris says from Scratch as well.

**Comment by kbower** [comment]
_**kbower** trac comment on Mar 12, 2013, 3:10:36 PM:_

Replying to [sjohnson](comment:16):
> Scott,
> ''[snip]'' As well, would this functionality allow returns from multiple split orders to one return tab? It would be great to see that a credit memo could only be keyed for merchandise on one split transportation at a time. This will clean up the back end with a credit memo amount not exceeding the amount of that specific delivery document number in CINQ.
> Thanks,
> Melissa
> 

We cannot support this because of tax implications.  (Perhaps later we can allow this provided the tax authority is the same, but for starting out, it will not be allowed.)

**Comment by kbower** [comment]
_**kbower** trac comment on Mar 12, 2013, 3:24:25 PM:_

Wayne,
    Typically, you seem to give the most complete answer, so I am hoping you will tell me all that needs to happen upon CRM creation and finalization.

I understand that creation of CRM is very very similar to creation of MCR:
 * No inventory interaction at this point at all (except maybe INV_XREF record created??? When does that occur?)
 * Tax calculated in the same way as MCR
 * Creation of 'A'nticipated Artran
 * Anything else?

Some differences:
 * We should have a permission that is required in order to add an item not on the original order (or create a Return with no original order).
 * DEL table:  How does scheduling work?  CRM's don't add actual points to the DEL record, I assume, correct? Please explain. (Including when you cannot schedule.  Can I still schedule CRM for closed DEL records?  I assume YES)
 * Must the SAL be finalized, or does GERS allow writing a Return against an Open SAL?
 * I can guess, but please explain this System Parameter: 34. If SKU is not on original sale or original sale is not on file, attach SPIFF amount to CRM line items in SOE & SOM? Y

Lastly, please explain how closing a return should occur.
 * The site and location need to be provided for each line, correct?
 * What if one of them never made it back?  Can you still close the CRM or must you first void the line that wasn't returned?
 * Can someone enter the location ahead of time and save that with the CRM still Open or can locations only be recorded when closing?
 * Inventory is created (essentially identical to unreserving the line to a different location, correct?)
   * Anything happen with out of carton logic?
   * INV_XREF record from PEN to FIF and document number/line deleted?
 * I presume the artran status goes from A to T?
 * What else happens?


Thanks very much, Wayne,
Kent

**Comment by kbower** [comment]
_**sal** trac comment on Mar 12, 2013, 5:27:54 PM:_

The debit and credit should be able to be tied to a contract number thus it would increase or decrease the balance on the contract. It should not affect the terms of the contract or recalculate any interest, etc.  I think the way GERS works when you are creating debits, credits and credit memos it asks for a contract number to associate it with which would set the AR type to "I" on the transaction.

>Comment (by kbower):


>Replying to [comment:22 sal]:
>> Be sure to plan for CRM/Returns associated with installment contracts. I  know that debits and credits were not setup to tie back to an installment  account and that needs to be fixed so maybe with returns it can be done  from the start.


>We never were able to definitively say what the correct approach is for  MCR/MDB tied to installment, or at least I was never made aware if that  was resolved what the approach should be.

----
_see [#2281](2281)_

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 5, 2013, 5:14:06 PM:_

Kent,

Good afternoon, first I want to apologize for the late response but it has been very busy around here with several projects going on. As promised I wanted to respond before the week has end however this will not be complete and I reserve the right at add, alter or drop any statement within this email in the future.

You are right in your assumptions that it is similar to creation of an MCR. INV_XREF record will be created when the credit memo is finalized and the item is brought back into inventory.

Correct CRM order should not add points to a DEL record. You cannot add a CRM if the delivery date is closed.

You can write a credit memo against an open sale but a security is needed.

If the item is not on the original sale or the original sale is not on file but the item has a spiff amount in ITM do you want to attach that spiff on SO_LN is what the system parameter is asking.

Closing a return:

Yes, the location should be added for every inventory sku.
You must void the line for unreturned items before finalizing.
The locations can only be recorded when finalizing.
Correct, inventory creation is identical.
I believe the system assumes it is automatically out of carton.
Correct on AR_TRN record.

If the customer has transfer of liability with the finance company your need to do the dollar transfers for financed credit memos.
You need t make entries into ITM_TRN for the returning merchandise.

I started writing the information below then proceeded to answer your questions as indicated above instead.

The following should happen on creation of a credit memo:

        1. There should be a security to restrict items to that of an existing order.
        2. there should be a security to allowing keying credit memos on open sales.
        3. If a person does not have the security to key a credit memo with no existing sales then we should do the following:
                a.        Have them reference the original sale
                b.        Return all items from that sale.
                c.        Remove items that are not being returned
                d.        Apply the same tax rate as on the existing sale (there should be a security to apply existing tax rate or current tax rate)
                e.        Finance and promotion information should be the same as the original sale.

Thanks,
 
Wayne Liburd

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 9, 2013, 11:36:24 AM:_

# Check list
 * ~~For CRM/MCR/MDB: list all lines (don't filter out fab like split orders)~~
 * Add ~~default return location &~~ valid return location flags
 * ~~DEL/PUP table~~:
   * ~~Change 'From Site' Label to 'Return Site'~~
   * ~~CRM order should not add points to a DEL record.~~ 
   * ~~You cannot add a CRM if the delivery date is closed.~~
   * ~~Delivery Calendar service for Returns~~
   * ~~Set date as today~~
 * ~~INV_XREF: I believe this occurs at line creation in status PEN (which differs from Wayne's report).~~ _moved to [#2295](2295)_
 * Finance and promotion information should be the same as the original sale. ??
 * ~~Return location to be shown~~, _see comment:9 & [#1984](1984)_
 * Closing:
   * Inventory creation, same as unreserve, except out of carton:
     * update the ITM_FIFL.OOC_QTY for out of carton
   * ITM_TRN records (CMF)
   * ArTran A to T
   * Transfer Liability
## Permissions
 * ~~Permission needed to add items to Return not on recorded sale.  (Same permission as being able to create a Return which adjusts no order.)~~  `'orderreturnnotsold'`
 * ~~Permission needed to create Return against still open Sale~~  `'orderreturnopensale'`
----
## System Parameters
 * Refund deliverycharge with return
 * Bring all lines over by default for CRM
 * (Bring all lines over by default for MCR)
----
## Behavior
 * ~~Do not allow to close return against unclosed adjusted order (spiff implications)~~
 * ~~For RF, extra checks needed~~ _moved to [#2295](2295)_
 * ~~SPIFF will be set as average spiff of outgoing sale,~~ _see comment:7_
   * ~~Reevaluate spiffs~~
 * ~~Warn / Error if tax rule added or changed to differ from adjusted order. (for all adjustments)~~

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 9, 2013, 12:10:59 PM:_

Wayne, would you agree that Loft should not allow users to close a credit
memo until the original order is closed first?  What about if the original
order is voided in the meantime?

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 9, 2013, 1:03:27 PM:_

Kent,

I agree that you should not be able to close a credit memo prior to the sale being closed first and also if the sale is voided.

Thanks,
Wayne Liburd

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 9, 2013, 3:41:41 PM:_

Wayne,
Would you please send the select statement for this system parameter:

    "34. If SKU is not on original sale or original sale is not on file, attach SPIFF amount to CRM line items in SOE & SOM? Y "

Thank you very much, as always!
Kent
----
Kent,

Good afternoon, the query is as follows:
 
SELECT SUBSTR(DATA, 208, 1)
FROM SYS_PARAMS
WHERE KEY = 'SALES1';

Thanks,
 
Wayne Liburd

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 18, 2013, 4:49:20 PM:_

(Changeset [[4039]](../commit/632cf09b6e3f8b227ab10b2de170cd0b983336f3))
* fixes [#2292](2292), Unnecessary rounding problems for Credits/Debits and Returns, also see #1037, (Returns)

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 23, 2013, 3:27:13 PM:_

When a return is finalized and the inventory is recreated:
 * does it come back to FIFL by default, I assume?
 * does outlet automatically come back as outlet?
 * what ITM_TRN is written?
 * does anything happen with out of carton? if so, what? does your company use out of carton?
 * is third party financing liability automatically transferred, or does that depend on how the CRM is written?

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 23, 2013, 4:47:09 PM:_

Mariam replying to [kbower](comment:31):
When a return is finalized and the inventory is recreated:
  * does it come back to FIFL by default, I assume?

        Yes, a record is created in FIFL

  * what ITM_TRN is written?

        CMF

  * does anything happen with out of carton? if so, what? does your company use out of carton?

        The only out of carton I know of is the one in ITEM. Since this is a record for the SKU, this is not affected. 
         We do not really use that field, instead we use the condition code in the wms.

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 24, 2013, 2:31:54 PM:_

Kent,

Yes, the inventory goes back to FIFL.

A CMF -  Credit Memo Finalization and UNR - Unreserved record transaction record is written to ITM_TRN.

The items should be listed as out of cartoon by default. We do use the out of cartoon.

Thanks,
 
Wayne Liburd

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 24, 2013, 3:12:53 PM:_

Hi Kent,
 
When an item is return the cost.
1. The first is the cost from the original sales order line. SO_LN, FIFO_CST  (if an original order is enter on the credit memo)
2. If no original order, GERS looks at existing costs ITM_FIFO and picks a cost.
3. If no original order and not ITM_FIFO records then the REPL_CST from ITM.
 
That is what I understand the pecking order.  I could test if need be.
 
ITM_TRN writes a record, ITT_CD is CMF, ITM_CD, FIFO_CST or CST, QTY, DEL_DOC_NUM, NEW_STORE_CD, NEW_LOC_CD, TRN_DT, EMP_CD..... are what you would think.  If need be I can pull some records for all the fields that get populated.
 
We done use "out of carton in GERS, in Dispatcher is where we track that.
 
Doug

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 24, 2013, 5:05:52 PM:_

Sal replying to [kbower](comment:31):
> When a return is finalized and the inventory is recreated:
>  * does it come back to FIFL by default, I assume?
>  * does outlet automatically come back as outlet?
>  * what ITM_TRN is written?
>  * does anything happen with out of carton? if so, what? does your company use out of carton?
>  * is third party financing liability automatically transferred, or does that depend on how the CRM is written?


1- yes
2- when you finalize the return you can designate if it's coming back into
outlet status or regular inventory
3- CRM
4- we use the WMS condition code for the out of carton stuff
5-it depends on how the CRM is written. If you designate it with a finance
company (which GERS does by default but you can take it out) then it will
transfer back when finalized, but sometimes you are doing a new sale and CRM
at the same time therefore you would not put the finance company in either
the CRM or the new SAL. That's at least how we currently do it.

**Comment by kbower** [comment]
_**kbower** trac comment on May 1, 2013, 3:26:44 PM:_

Mark,
    Is the ITM_INV record created for Returns when the CRM line is first written, in some sort of pending state, or is the ITM_INV record created when the return is finalized?  (I presume the latter, but please let me know.)

Thanks!
Kent
----
No.  Nothing is created in ITM_INV until finalized.

When it finalizes, the ITM_INV table is populated like this.

The OOC qty causes some problems if the field changes to a qty. (something quirky GERS does because of the assumption of it being out of original carton when we don’t use that functionality)

<image002.jpg>


And it also creates the ITM_INV2CST

<image009.jpg>


I looked into things a little further for where the cost comes from.

When the original order is referenced, it brings in the original cost from the SAL.

If there is no original document referenced, it brings it in at replacement cast from ITM

**Comment by kbower** [comment]
_**kbower** trac comment on May 2, 2013, 9:22:45 AM:_

Doug,
I just realized that GERS doesn't even copy the FIFO_CST from the original
SAL line when it creates the CRM line!  It copies the FIFO_DT and FIFL_DT,
but not the cost.

Can you answer or do an experiment to answer what happens if the SAL
SO_LN's FIFO_CST is completely unique (doesn't exist in ITM_FIFO) when the
return is finalized?  When creating the inventory, does GERS go back and
look up the FIFO_CST from the SAL line or does it look in existing
inventory for the FIFO_DT and FIFL_DT?  If it uses the SAL line's FIFO_CST,
then my big question is this: how does it keep track of which items from
the original SAL it already returned?   If there are several lines of the
same product on the original SAL with differing costs and more than one CRM
is written, how does it track this, or does it mess up and reuse the same
cost sometimes?

**Comment by kbower** [comment]
_**kbower** trac comment on May 2, 2013, 9:47:13 AM:_

Another issue is that if GERS allows a security to write a credit memo
against an open order, the original line may not yet be filled, or it may
be unfilled and refilled with a different piece that could make the CRM
line's FIFO_DT and FIFL_DT incorrect.

My guess is that GERS doesn't prevent this error, does it?  And if, at
finalization time for the return, there is no FIFO_DT and FIFL_DT (because
the CRM was written before the sale line was filled), would GERS just go
through the logic as if it was written against no line at all (as if no
original order) or would it still grab a (possibly) random cost from the
SAL, even though there is no FIFO_DT and FIFL_DT?

(Sorry for all the questions... Thank you!)

**Comment by kbower** [comment]
_**kbower** trac comment on May 2, 2013, 11:26:20 AM:_

_to Mark Bracey:_
Thank you for this information. Another retailer does use OOC... Do you have more specifics as to the issue you referenced regarding this?  I need to increment that qty apparently, but don't wish to cause you issues. 

Kent
----
Periodically they will show as an RF mismatch where one or the other tables,  ITM_INV or INV_XREF, will have the qty in the OOC field.  I remove the qty in either table to make them match.

As long as the 2 tables update the same, it makes no difference to us.  We just don’t use the functionality.

**Comment by kbower** [comment]
_**kbower** trac comment on May 7, 2013, 3:44:32 PM:_

Mark,

Incidentally it looks like it is using sequence MISC.SEQ_RF_ID (last value 2123955) and then adding a check digit to the end of it when it needs a new RF ID.

Any ideas on the specifications of the check digit that is at the end of the RF_IDs?  (How it is generated?)



On 5/7/2013 3:35 PM, Kent Bower wrote:

[[Image(untitled-part.jpe)]]

**Comment by kbower** [comment]
_**kbower** trac comment on May 7, 2013, 3:50:38 PM:_

Wow, check this out. Writing that CRM and voiding it moved the new INV_XREF to _ARC, so now there are two finalized RF_IDs showing as having been delivered to the customer, even though only one actually shipped.


```
SQL> select RF_ID_CD, ITM_CD, CREATE_DT, LAST_ACTN_DT, DISPOSITION, QTY, ORIGIN_CD from inv_xref_arc where del_doc_num = '0928207NMPG';

RF_ID_CD   ITM_CD    CREATE_DT            LAST_ACTN_DT DIS        QTY ORIGI
---------- --------- -------------------- -------------------- --- ---------- -----
0017536886 008164573 11-NOV-2011 00:00:00 07-MAY-2013 00:00:00 FIN          1 SOM
0021239374 008164573 07-MAY-2013 00:00:00 07-MAY-2013 00:00:00 FIN          1 SOM
```

**Comment by kbower** [comment]
_**kbower** trac comment on May 9, 2013, 2:23:52 PM:_

```
svn merge https://svn.retailarchitects.com/repos/rms/branches/returns_b@4136 https://svn.retailarchitects.com/repos/rms/branches/returns_b@4137 .
```

**Comment by kbower** [comment]
_**kbower** trac comment on May 9, 2013, 2:44:16 PM:_

(Changeset [[4121]](../commit/8c4ca92fed4cad41d1bdbe295edea9af26b104ab))
* changes to ease database migrations due to returns branch in preparation for return, see #1037

**Comment by kbower** [comment]
_**kbower** trac comment on Jun 4, 2013, 12:47:34 PM:_

(Changeset [[4211]](../commit/2a53cf8298c41a40d5e53e116953286b4a843697))
* Implement Returns (Phase 1), closes #1037 #all
   * What is NOT implemented with this phase:
     * Financing implications with returns ([#2224](2224))
     * Writing returns against open order with RF enabled ([#2295](2295))
     * Writing returns for more than was originally sold with RF enabled ([#2295](2295))
     * Returns with an Outlet status different from original line ([#2327](2327))
     * Closing return lines to new NAS or outlet status ([#2327](2327), [#2328](2328))
     * Closing returns for more than was originally sold (figuring cost issue) ([#2329](2329))
     * Closing returns with V12 ([#2317](2317))
     * Closing returns with RF ([#2295](2295))
     * Linking to installment contracts, ([#2281](2281))
     * Writing returns with no adjusted order (from scratch), ([#2291](2291))

**Comment by kbower** [comment]
_**kbower** trac status change on Jun 4, 2013, 12:47:34 PM:_ **closed**
